pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,128 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
```

Comment:
I was looking for a way to reverse a slice, searched the standard library, noticed the symbol `Reverse` in the `sort` package, glanced at the usage example, and off I went, not noticing—believe it or not— that using `sort` does in fact _sort the input_, which was not my intention.

Given that:
```go
unc reverseStrings(s []string) {
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		s[i], s[j] = s[j], s[i]
	}
}

func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
	components := strings.Split(path, ".")
/* What should happen if the path is empty?
	if len(components) == 0 {
		return nil
	}
*/
	reverseStrings(components)
	m := data
	fmt.Println("components:", components)
	for _, c := range components {
		m = map[string]interface{}{c: m}
	}
	return m
}
```